src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -115,4 +127,263 @@ private boolean isSeqColL1Present(SeqColLevelOneEntity levelOneEntity) {
         Optional<SeqColLevelOneEntity> existingSeqCol = levelOneService.getSeqColLevelOneByDigest(levelOneEntity.getDigest());
         return existingSeqCol.isPresent();
     }
+
+    /**
+     * Compare two seqCol L2 objects*/
+    public SeqColComparisonResultEntity compareSeqCols(
+            String seqColADigest, SeqColLevelTwoEntity seqColAEntity, String seqColBDigest, SeqColLevelTwoEntity seqColBEntity) throws InvocationTargetException, IllegalAccessException {
+
+        SeqColComparisonResultEntity comparisonResult = new SeqColComparisonResultEntity();
+
+        // "digests" attribute
+        comparisonResult.putIntoDigests("a", seqColADigest);
+        comparisonResult.putIntoDigests("b", seqColBDigest);
+
+        // "arrays" attribute
+        setArraysAttribute(seqColAEntity, seqColBEntity, comparisonResult);
+
+        // "elements"
+        List<String> seqColALengths = seqColAEntity.getLengths();
+        List<String> seqColBLengths = seqColBEntity.getLengths();
+        List<String> seqColANames = seqColAEntity.getNames();
+        List<String> seqColBNames = seqColBEntity.getNames();
+        List<String> seqColASequences = seqColAEntity.getSequences();
+        List<String> seqColBSequences = seqColBEntity.getSequences();
+
+        // "elements" attribute | "total"
+        setTotalElementsAttribute(seqColAEntity, seqColBEntity, comparisonResult);
+
+        // "elements" attribute | "a-and-b"
+        setA_and_B_ElementsAttribute(seqColAEntity, seqColBEntity, comparisonResult);
+
+        // "elements" attribute | "a-and-b-same-order"
```

Comment:
Why shouldn't the logic below refactored into a separate function as well similar to the above 2 functions?

Reply:
In the code below we assume that we already know the fields that are present in each seqCol objects.
Which might not be the case all the time, as some seqCol might contain additional attributes in the future. That's why I wanted to make it generic.